moveit_core/trajectory_processing/src/ruckig_traj_smoothing.cpp

```diff
@@ -132,61 +117,142 @@ bool RuckigSmoothing::applySmoothing(robot_trajectory::RobotTrajectory& trajecto
   }
 
   ruckig::Result ruckig_result;
-  for (size_t waypoint_idx = 0; waypoint_idx < num_waypoints - 1; ++waypoint_idx)
+  bool smoothing_complete = false;
+  robot_trajectory::RobotTrajectory original_trajectory_copy(trajectory);
+  size_t duration_extension_attempts = 0;
+  while (rclcpp::ok() && (duration_extension_attempts < MAX_DURATION_EXTENSION_ATTEMPTS) && !smoothing_complete)
   {
-    moveit::core::RobotStatePtr next_waypoint = trajectory.getWayPointPtr(waypoint_idx + 1);
+    for (size_t waypoint_idx = 0; waypoint_idx < num_waypoints - 1; ++waypoint_idx)
+    {
+      moveit::core::RobotStatePtr next_waypoint = trajectory.getWayPointPtr(waypoint_idx + 1);
 
-    getNextCurrentTargetStates(ruckig_input, ruckig_output, next_waypoint, num_dof, idx);
+      getNextCurrentTargetStates(ruckig_input, ruckig_output, next_waypoint, num_dof, idx);
```

Comment:
This function signature and name is very confusing. `ruckig_input` is the output and `ruckig_output` the actual input. Can you make `ruckig_output` const since it's not changed (it's an input), move `ruckig_input` to the last position and rename the function to something simpler like `getNextRuckigInput()`? See [here](https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs) for style guide reference.

Reply:
LOL, thanks for the suggestion. Will do. It's inherently confusing b/c the previous output is fed back in as input.

Reply:
Also please be aware I wrote this PR very quickly so I'm not too surprised the style isn't perfect...